stl/inc/future

```diff
@@ -725,21 +727,20 @@ template <class _Ty>
 class _State_manager {
     // class for managing possibly non-existent associated asynchronous state object
 public:
-    _State_manager() : _Assoc_state(nullptr) { // construct with no associated asynchronous state object
-        _Get_only_once = false;
+    _State_manager() noexcept
+        : _Assoc_state(nullptr), _Get_only_once(false) { // construct with no associated asynchronous state object
     }

```

Comment:
No change requested, but I note that in our modern coding style we'd make these default member initializers and `= default` this constructor (letting the compiler make it implicitly `noexcept`). (This would let us clean up a lot of `_Assoc_state(nullptr)` mem-initializers.)

Reply:
I considered default member initializers for this but gave up the changes, because `_Get_only_once` might be writen 3 times in the copy/move ctors. Perhaps this should be done in some further refactoring.

Reply:
Now I think such further refactoring is worthy and trying to do it.